src/vs/workbench/contrib/terminal/browser/terminalTypeAheadAddon.ts

```diff
@@ -845,7 +865,7 @@ export class PredictionTimeline {
 					this.succeededEmitter.fire(prediction);
 					this.lookBehind = prediction;
 					this.expected.shift();
-					break;
+					break ReadLoop;
```

Comment:
this seems wrong...

Reply:
that is 😛 it could leave unread data in the buffer which would cause us to invalidate the rest of the predictions (in `if (!reader.eof)`

Reply:
the problem is... fish and PowerShell have stuff like this:
![image](https://user-images.githubusercontent.com/2644648/106967226-a9693180-66fb-11eb-8681-b838d6d02216.png)

so when we just `break` and not `break ReadLoop` it ends up failing because there are extra characters after even though what was predicted was "correct"

Reply:
Ugh. Maybe we modify the eof check to emit the remainder of the string, and only invalidate _if_ the cursor is not returned to the expected `physicalCursor` position at the end of that progress.

Reply:
@TylerLeonhardt A side question: is that dark-gray part of the command-line a history autocomplete? And if yes, how to configure it, please? I used to have it but somehow managed to lose it and I cannot remember how to configure it back (or even how it's called).

Reply:
Take a look at the blog post here 😁:

https://devblogs.microsoft.com/powershell/announcing-psreadline-2-1-with-predictive-intellisense/

Reply:
@connor4312 what eol logic are you referring to?

Reply:
ugh we also have to handle this for backspace...